vendor/bat-native-ads/src/bat/ads/internal/frequency_capping/exclusion_rules/daypart_frequency_cap_unittest.cc

```diff
@@ -0,0 +1,291 @@
+  daypart_info.end_minute = current_time - 60;
+  ad.dayparts.push_back(daypart_info);
+
+  CatalogDaypartInfo daypart_info_2;
+  daypart_info_2.days_of_week = tomorrow_dow;
+  daypart_info_2.start_minute = current_time + 120;
+  daypart_info_2.end_minute = current_time + 180;
+  ad.dayparts.push_back(daypart_info_2);
+
+  CatalogDaypartInfo daypart_info_3;
+  daypart_info_3.days_of_week = current_dow;
+  daypart_info_3.start_minute = current_time - 60;
+  daypart_info_3.end_minute = current_time + 60;
+  ad.dayparts.push_back(daypart_info_3);
+
+  // Act
+  const bool should_exclude = frequency_cap_->ShouldExclude(ad);
+
+  // Assert
+  EXPECT_FALSE(should_exclude);
+}
+
+TEST_F(BatAdsDaypartFrequencyCapTest, DisallowIfNoMatches) {
+  // Arrange
+  CreativeAdInfo ad;
+  ad.creative_set_id = kCreativeSetId;
+
+  base::Time::Exploded exploded;
+  base::Time::Now().LocalExplode(&exploded);
+  int current_time = 60 * exploded.hour + exploded.minute;
+  std::string tomorrow_dow =
+      base::NumberToString((exploded.day_of_week + 1) % 7);
+  std::string current_dow = base::NumberToString(exploded.day_of_week);
+
+  CatalogDaypartInfo daypart_info;
+  daypart_info.days_of_week = tomorrow_dow;
+  daypart_info.start_minute = current_time - 60;
+  daypart_info.end_minute = current_time - 60;
+  ad.dayparts.push_back(daypart_info);
+
+  CatalogDaypartInfo daypart_info_2;
+  daypart_info_2.days_of_week = tomorrow_dow;
+  daypart_info_2.start_minute = current_time + 120;
+  daypart_info_2.end_minute = current_time + 180;
+  ad.dayparts.push_back(daypart_info_2);
+
+  CatalogDaypartInfo daypart_info_3;
+  daypart_info_3.days_of_week = current_dow;
+  daypart_info_3.start_minute = current_time + 60;
+  daypart_info_3.end_minute = current_time + 120;
+  ad.dayparts.push_back(daypart_info_3);
+
+  // Act
+  const bool should_exclude = frequency_cap_->ShouldExclude(ad);
+
+  // Assert
+  EXPECT_TRUE(should_exclude);
+}
+
+
+TEST_F(BatAdsDaypartFrequencyCapTest, DisallowIfWrongDay) {
+  // Arrange
+  CreativeAdInfo ad;
+  ad.creative_set_id = kCreativeSetId;
+
+  base::Time::Exploded exploded;
+  base::Time::Now().LocalExplode(&exploded);
+  int current_time = 60 * exploded.hour + exploded.minute;
+
+  // Go to next day
+  std::string tomorrow_dow =
+      base::NumberToString((exploded.day_of_week + 1) % 7);
+
+  CatalogDaypartInfo daypart_info;
+  daypart_info.days_of_week = tomorrow_dow;
+  daypart_info.start_minute = current_time - 60;
+  daypart_info.end_minute = current_time - 60;
+  ad.dayparts.push_back(daypart_info);
+
+  // Act
+  const bool should_exclude = frequency_cap_->ShouldExclude(ad);
+
+  // Assert
+  EXPECT_TRUE(should_exclude);
+}
+
+TEST_F(BatAdsDaypartFrequencyCapTest, DisallowIfWrongHours) {
+  // Arrange
+  CreativeAdInfo ad;
+  ad.creative_set_id = kCreativeSetId;
+
+  base::Time::Exploded exploded;
+  base::Time::Now().LocalExplode(&exploded);
+  int current_time = 60 * exploded.hour + exploded.minute;
```

Comment:
Const

Reply:
Please use constant from base::Time instead of 60

Reply:
done